starter-code/pizza.js

```diff
@@ -1,85 +1,184 @@
   pepperonni: true,
   mushrooms: true,
   greenPeppers: true,
   whiteSauce: false,
   glutenFreeCrust: false
-}
+};
 
 // This function takes care of rendering the pizza based on the state
 // This function is triggered once at the begining and everytime the state is changed
+//この関数は、状態に基づいてピザをレンダリングします
+//この関数は、開始時に一度、状態が変更されるたびにトリガーされます
 function renderEverything() {
-  renderPepperonni()
-  renderMushrooms()
-  renderGreenPeppers()
-  renderWhiteSauce()
-  renderGlutenFreeCrust()
-
-  renderButtons()
-  renderPrice()
+  renderPepperonni();
+  renderMushrooms();
+  renderGreenPeppers();
+  renderWhiteSauce();
+  renderGlutenFreeCrust();
+
+  renderButtons();
+  renderPrice();
 }
 
 function renderPepperonni() {
-  document.querySelectorAll('.pep').forEach(function($pep){
+  document.querySelectorAll(".pep").forEach(function($pep) {
     if (state.pepperonni) {
       $pep.style.visibility = "visible";
-    }
-    else {
+    } else {
       $pep.style.visibility = "hidden";
     }
-  })
+  });
 }
 
 function renderMushrooms() {
+  document.querySelectorAll(".mushroom").forEach(function(mushroom) {
+    if (state.mushrooms) {
+      mushroom.style.visibility = "visible";
+    } else {
+      mushroom.style.visibility = "hidden";
+    }
+  });
   // Iteration 1: set the visibility of `<section class="mushroom">`
 }
 
 function renderGreenPeppers() {
+  document.querySelectorAll(".green-pepper").forEach(function(greenPepper) {
+    if (state.greenPeppers) {
+      greenPepper.style.visibility = "visible";
+    } else {
+      greenPepper.style.visibility = "hidden";
+    }
+  });
   // Iteration 1: set the visibility of `<section class="green-pepper">`
 }
 
 function renderWhiteSauce() {
+  document.querySelectorAll(".sauce").forEach(function(white) {
+    if (state.whiteSauce) {
+      white.classList.add("sauce-white");
+    } else {
+      white.classList.remove("sauce-white");
+    }
+  });
   // Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
 }
 
 function renderGlutenFreeCrust() {
+  document.querySelectorAll(".crust").forEach(function(crustFree) {
+    if (state.glutenFreeCrust) {
+      crustFree.classList.add("crust-gluten-free");
+    } else {
+      crustFree.classList.remove("crust-gluten-free");
+    }
+  });
+
   // Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
 }
 
+document.querySelectorAll("active").onclick = function() {
+  state.glutenFreeCrust = !state.glutenFreeCrust;
+  renderEverything();
+};
+
 function renderButtons() {
-  // Iteration 3: add/remove the class "active" of each `<button class="btn">`
+  /*
+  document.getElementsByClassName(".btn").forEach(function(button) {
+    button.onclick = toggleButton;
+  });
+
+  function toggleButton() {
+    const currenButton = event.target;
+    currentButton.classList.toggle("btn.active");
+  }
+  // Iteration 3: add/remove the class "active" of each `<button class="btn">`*/
+
+  if (state.pepperonni) {
```

Comment:
Hey Saeka, were you here the other day when I showed a way of doing the prices?
You could have used the same logic for there as well: 
```
function renderButtons() {
  for (const key in ingredientClasses) {
    if (state[key]) {
      document
        .querySelector(`.btn-${ingredientClasses[key]}`)
        .classList.add("active");
    } else
      document
        .querySelector(`.btn-${ingredientClasses[key]}`)
        .classList.remove("active");
  }
}
```